Tools/clinic/clinic.py

```diff
@@ -4755,8 +4772,8 @@ def to_required(self):
         """
         Transition to the "required" parameter state.
         """
-        if self.parameter_state != self.ps_required:
-            self.parameter_state = self.ps_required
+        if self.parameter_state != ParamState.REQUIRED:
```

Comment:
Each member in an enum is conceptually a singleton, so it's idiomatic to use identity checks rather than equality checks

```suggestion
        if self.parameter_state is not ParamState.REQUIRED:
```

Reply:
Good idea.